CMakeLists.txt

```diff
@@ -118,3 +118,16 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
 # Enable unit testing by including the `test` subdirectory.
 enable_testing()
 add_subdirectory(test)
+
+###############
+## UNINSTALL ##
+###############
+
+# Add uninstall target
+configure_file(
+  "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/cmake/cmake_uninstall.cmake"
+  IMMEDIATE @ONLY)
+add_custom_target(uninstall
+  "${CMAKE_COMMAND}" -P
+  "${CMAKE_CURRENT_BINARY_DIR}/cmake/cmake_uninstall.cmake")
```

Comment:
Like @psigen and I discussed before, the `uninstall` target will only work as expected if the `build` directory is present and has not changed since you ran `make install`. I would prefer to install the `cmake_uninstall.cmake` script as part of the installation step.